core/datastore/src/main/java/com/sseotdabwa/buyornot/core/datastore/UserPreferencesDataSourceImpl.kt

```
@@ -22,21 +27,41 @@ class UserPreferencesDataSourceImpl
         override val preferences: Flow<UserPreferences> =
-            context.dataStore.data.map { prefs ->
+            context.userPreferencesDataStore.data.map { prefs ->
                 UserPreferences(
                     displayName = prefs[Keys.DISPLAY_NAME] ?: UserPreferences().displayName,
                     accessToken = prefs[Keys.ACCESS_TOKEN] ?: UserPreferences().accessToken,
                     refreshToken = prefs[Keys.REFRESH_TOKEN] ?: UserPreferences().refreshToken,
+                    userType =
+                        prefs[Keys.USER_TYPE]?.let {
+                            try {
+                                UserType.valueOf(it)
+                            } catch (e: IllegalArgumentException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)